lib/bank/account.rb

```diff
@@ -0,0 +1,37 @@
+module Bank
+  class Account
+    # attr_accessor creates an implicit balance method
+    attr_accessor :id, :balance
+
+    def initialize(id, initial_balance)
+      raise ArgumentError, "Negative balance is invalid." if initial_balance.to_i < 0
+
+      @id = id
+      @balance = initial_balance.to_i
+      @minimum_balance = 0
+    end
+
+    def withdraw(amount)
+      # Integer() will turn strings of numbers into valid integers.
+      # If it's not a whole number or includes invalid characters
+      # it'll return 0 instead
+      amount = Integer(amount) rescue 0
```

Comment:
Using `Integer()` is a good way of dealing with parsing numbers into strings. I'm a little concerned about the rescue, because you're triggering something known as a silent failure.

So if someone tries to `withdraw('eleventy billion dollars')`, this will fail, default the withdraw amount to 0, and return the original balance. But there is no indication on why the withdrawal failed. So in this case, an `ArgumentError` would be appropriate:
```
begin
  amount = Integer(amount)
rescue
  raise ArgumentError, 'Withdraw amount not a number'
end
```